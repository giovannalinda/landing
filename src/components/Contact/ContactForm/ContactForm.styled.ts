import styled, { css } from 'styled-components'
import { theme } from '~/styles'

import { Container as Button } from '../../Button/Button.styled'

export const globalInputCss = css`
  border: 1px solid ${theme.colors.gray100};
  width: 100%;
  min-height: 5.2rem;
  border-radius: 0.4rem;
  padding: 1.4rem 2rem 1.4rem 2rem;
  transition: 0.3s;
  outline: 0;

  &:focus {
    border-color: ${theme.colors.secondaryPurple};
  }

  &::placeholder {
    color: ${theme.colors.gray500};
  }
`

export const Form = styled.form`
  display: grid;
  grid-gap: ${theme.space[4]};
`

export const FormGroup = styled.div`
  display: grid;
  grid-gap: ${theme.space[4]};
  grid-template-columns: 1fr 1fr;
`

export const Input = styled.input`
  ${globalInputCss}
`

export const TextArea = styled.textarea`
  resize: none;
  font-family: ${theme.font.family};

  ${globalInputCss};
`

export const SendMessage = styled(Button)`
  display: flex;
  align-items: center;
  font-weight: ${theme.font.weigths.semiBold};
  margin-top: 1rem;
  margin-right: auto;

  svg {
    margin-left: ${theme.space[2]};
  }
`
