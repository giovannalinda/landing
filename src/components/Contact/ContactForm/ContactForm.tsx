import { FiSend } from 'react-icons/fi'
import { useTranslation } from '~/lib/useTranslation'

import * as S from './ContactForm.styled'

export function ContactForm() {
  const { t } = useTranslation()

  return (
    <S.Form>
      <S.FormGroup>
        <S.Input type='text' placeholder={t('Full name')} />
        <S.Input type='email' placeholder={t('Email adress')} />
      </S.FormGroup>
      <S.Input type='text' placeholder={t('Subjects')} />
      <S.TextArea placeholder={t('Message')} rows={7} />
      <S.SendMessage variant='purple' size='md' type='submit'>
        {t('Send message')} <FiSend size={20} />
      </S.SendMessage>
    </S.Form>
  )
}
