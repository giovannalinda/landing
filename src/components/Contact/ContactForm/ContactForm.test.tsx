import { render } from '@testing-library/react'
import { ContactForm } from './ContactForm'

describe('ContactForm', () => {
  it('should match snapshot', () => {
    const { container } = render(<ContactForm />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
