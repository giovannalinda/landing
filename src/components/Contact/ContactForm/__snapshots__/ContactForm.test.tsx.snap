// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should match snapshot 1`] = `
<form
  class="sc-dkzDqf dfzxQ"
>
  <div
    class="sc-hKMtZM gTrNLx"
  >
    <input
      class="sc-eCYdqJ kciHRe"
      placeholder="Full name"
      type="text"
    />
    <input
      class="sc-eCYdqJ kciHRe"
      placeholder="Email adress"
      type="email"
    />
  </div>
  <input
    class="sc-eCYdqJ kciHRe"
    placeholder="Subjects"
    type="text"
  />
  <textarea
    class="sc-jSMfEi bBtODK"
    placeholder="Message"
    rows="7"
  />
  <button
    class="sc-bczRLJ sc-gKXOVf cQHTfc dSmJlC"
    type="submit"
  >
    SEND MESSAGE 
    <svg
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="22"
        x2="11"
        y1="2"
        y2="13"
      />
      <polygon
        points="22 2 15 22 11 13 2 9 22 2"
      />
    </svg>
  </button>
</form>
`;
